src/Riok.Mapperly/Descriptors/DescriptorBuilder.cs

```diff
@@ -123,9 +127,29 @@ private void ExtractUserMappings()
                 userMapping.TargetType
             );
 
+            if (userMapping.Method is { IsStatic: true, IsPartialDefinition: true })
+            {
+                _mapperDescriptor.Static = true;
+            }
+            else if (firstNonStaticUserMapping is null && userMapping.Method is { IsStatic: false })
```

Comment:
Nit: For single property conditions and null checks this code base usually doesn't use pattern matching: `firstNonStaticUserMapping == null && !userMapping.Method.IsStatic`